pkg/pipeline/common_steps.go

```diff
@@ -79,18 +78,25 @@ func updateObject(obj PipelineObject, data *ExecutionContext) ExecutionResult {
 		}
 	}
 
-	updatedObj := rtObj.DeepCopyObject()
-	err := data.Client.Update(context.TODO(), updatedObj)
+	if !data.Deleted {
+		err := data.Client.Status().Update(context.TODO(), rtObj)
```

Comment:
Wont' we have the same problem here, that `rtObj` will be updated with the current state of it in the cluster?

Or does this call only update the `.status` sub-resource?

Reply:
This only updates the subresource.